### PR TITLE
Harden analysis server integration

### PR DESCRIPTION
**Pull request by lukechurch**
@devoncarew 

This PR makes two changes

- Replace spin locks with a scheduler that can do time outs
- Uses this to deal with the case where the Analysis Server never completes some requests

It also contains a bulk reformat to use the formatter - I should have done in a separate PR, sorry about that, but the impact isn't too bad.

In the very long running test suite these correctly result in the Analysis Server self-healing. The thing that causes it to die appears to be a state dependent issue rather than a single code change issue. I can currently only reproduce the case after around 30 hours of usage - this doesn't fix that problem it just bounds the outage.

I will follow up with the AS team about getting the underlying bugs fixed.

cc: @danrubel

### CODE REVIEW

File: lib/src/analysis_server.dart [hunk 1]
```diff
@@ -32,12 +33,17 @@ final _WARMUP_SRC_HTML = "import 'dart:html'; main() { int b = 2;  b++;   b. }";
 final _WARMUP_SRC = "main() { int b = 2;  b++;   b. }";
 final _SERVER_PATH = "bin/_analysis_server_entry.dart";
 
+// Use very long timeouts to ensure that the server has enough time to restart.
+const int _ANALYSIS_SERVER_TIMEOUT_S = 15;
```
Comment (devoncarew) [inline lib/src/analysis_server.dart]: recommend to use `Duration` here instead -

File: lib/src/analysis_server.dart [hunk 2]
```diff
@@ -47,20 +53,18 @@ class AnalysisServerWrapper {
...
     serverConnection = new _Server(this.sdkPath);
+    serverScheduler = new scheduler.TaskScheduler();
   }
 
   Future<api.CompleteResponse> complete(String src, int offset) async =>
-    completeMulti({"main.dart" : src},
-      new api.Location()
-        ..sourceName = "main.dart"
-        ..offset = offset);
+      completeMulti({"main.dart": src}, new api.Location()
```
Comment (devoncarew) [inline lib/src/analysis_server.dart]: here and below - perhaps prefer to not use the function shorthand when the function spans across several lines (even if it is only one expression)?

**Comment by devoncarew** [comment]
lgtm! Is the build failure (re: the benchmarks) something you can repo locally, or is it transient?

File: lib/src/analysis_server.dart (hunk 1 above)
Comment (lukechurch) [inline lib/src/analysis_server.dart]: Yeah, good point. Will do.

File: lib/src/analysis_server.dart (hunk 2 above)
Comment (lukechurch) [inline lib/src/analysis_server.dart]: ok.

**Comment by lukechurch** [comment]
I think the build failure is transient, it doesn't occur locally for me. If it repeats on the push to address your other comments I'll dig into it a bit further.

File: lib/src/analysis_server.dart (hunk 2 above)
Comment (lukechurch) [inline lib/src/analysis_server.dart]: done. This feels better and is more consistent with the other methods.

**Pull request merged**